Source/Core/VideoBackends/Vulkan/VKTexture.cpp

```diff
@@ -794,6 +894,55 @@ void VKStagingTexture::CopyFromTexture(const AbstractTexture* src,
   m_flush_fence_counter = g_command_buffer_mgr->GetCurrentFenceCounter();
 }
 
+void VKStagingTexture::CopyFromTextureToLinearImage(const VKTexture* src_tex,
+                                                    const MathUtil::Rectangle<int>& src_rect,
+                                                    u32 src_layer, u32 src_level,
+                                                    const MathUtil::Rectangle<int>& dst_rect)
+{
+  // The proprietary Qualcomm driver allocates a temporary image when copying from an image
+  // with optimal tiling to a buffer.
+  // That allocation is very slow, so we just do it ourself and reuse the intermediate image.
```

Comment:
What do you mean by "with optimal tiling"? Does it not allocate a temporary image when copying from an image that isn't tiled optimally (and if so, what exactly does that mean?)

Reply:
VkImages created with VK_TILING_LINEAR rather than VK_TILING_OPTIMAL. Basically it's linear rather than using some implementation dependent swizzling. That doesn't really matter too much for us, the interesting part is that copying from those images doesn't do an allocation + blit.

https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkImageTiling.html

Reply:
Ah, ok. Maybe phrase it as "with optimal tiling (VK_IMAGE_TILING_OPTIMAL) to a buffer" instead?